apache/trafficcontrol
# Update Traffic Router Java version to 11

Closes: #3196

This PR makes Traffic Router use Java 11 by default for compilation, testing, and running.

Big thanks to @ARMmaster17 for getting Traffic Router itself to work using Java 11. Besides changing Java versions around the project, this PR mainly just updates PMD to a version that supports targeting Java 11, which involved fixing new PMD "failures" that PMD 6.29.0 (up from PMD 5.3.5 in `maven-pmd-plugin` version 3.6) finds.
<hr>

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Router
- Automation (GitHub Actions, Ansible Roles)

## What is the best way to verify this PR?

- Run tests using Java 11
    ```shell
    docker-compose -f traffic_router/tests/docker-compose.yml up --build unit_router
    ```
- Compile Traffic Router using Java 11:
    ```shell
    ./pkg -bv traffic_router_build
    ```
- Verify that Traffic Router, including the resource watchers, work as expected by running the TR external and integration tests

## PR submission checklist
